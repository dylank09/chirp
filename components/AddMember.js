import React, { useState } from "react";
import { StyleSheet, ScrollView, Text, View, TextInput } from "react-native";

import { theme } from "../assets/Theme";
import TextAlert from "../components/TextAlert";
import ChirpButton from "./ChirpButton";

import firebase from "firebase/app";
import "firebase/firestore";
import { useCollectionData } from "react-firebase-hooks/firestore";
import app from "../config/FirebaseConfig";

const firestore = firebase.firestore(app);

export default function AddMember({ currentMembers, chatId, projectId }) {
  const [email, setEmail] = useState("");
  const [addMemberAlert, setAddMemberAlert] = useState("");

  const fsRef = chatId
    ? firestore.collection("chatGroups").doc(chatId)
    : firestore.collection("projects").doc(projectId);
  const usersRef = firestore.collection("users");
  const query = usersRef.where("email", "==", email);
  const [user] = useCollectionData(query, { idField: "userid" });

  function addMember() {
    setAddMemberAlert("");

    if (!user[0]) {
      setAddMemberAlert("No user with that email exists");
      return;
    }

    let userEmail = user[0].email;
    if (!currentMembers.includes(userEmail)) {
      let newMembers = currentMembers;
      newMembers.push(userEmail);
      fsRef.update({
        members: newMembers,
      });
      setEmail("");
      setAddMemberAlert("Successful");
    } else {
      setAddMemberAlert("User is already in the chat");
    }
  }

  return (
    <View style={styles.container}>
      <Text style={styles.optionsText}>Add member</Text>
      <View style={styles.addMember}>
        <TextInput
          style={styles.memberTextBox}
          placeholder="Email"
          placeholderTextColor={theme.colors.placeholderColor}
          value={email}
          onChangeText={setEmail}
        ></TextInput>
        <ChirpButton
          text="Add"
          onPress={() => addMember()}
          width="25%"
        ></ChirpButton>
      </View>
      <TextAlert text={addMemberAlert}></TextAlert>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    width: "100%",
    marginBottom: 20,
    borderBottomColor: theme.colors.jet,
    borderBottomWidth: 1,
  },
  optionsText: {
    color: theme.colors.text,
    fontSize: theme.dimensions.standardFontSize,
    fontWeight: "bold",
    alignSelf: "center",
    marginTop: 4,
  },
  addMember: {
    borderRadius: 18,
    borderColor: theme.colors.primary,
    borderWidth: 1,
    flexDirection: "row",
    margin: 7,
    width: "90%",
    justifyContent: "space-around",
    alignSelf: "center",
  },
  memberTextBox: {
    width: "60%",
    color: "white",
    fontSize: theme.dimensions.standardFontSize + 2,
    margin: 7,
  },
});
