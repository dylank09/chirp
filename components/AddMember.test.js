import React from "react";
import renderer from "react-test-renderer";
// import { render } from "@testing-library/react-native";

import AddMember from "./AddMember";

describe("<AddMember />", () => {
  const tree = renderer.create(<AddMember />).toJSON();

  it("renders correctly", () => {
    expect(tree).toMatchSnapshot();
  });

  //   it("user is present", () => {
  //     const { getByTestId } = render(<AddMember user="Testing User" />);
  //     const foundUser = getByTestId("user");
  //     expect(foundUser).toBeTruthy();
  //   });
});
