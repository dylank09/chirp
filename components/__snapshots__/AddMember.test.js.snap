// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddMember /> renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 20,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "rgb(255, 255, 255)",
        "fontSize": 15,
        "fontWeight": "bold",
        "marginTop": 4,
      }
    }
  >
    Add member
  </Text>
  <View
    style={
      Object {
        "alignSelf": "center",
        "borderColor": "rgb(193, 73, 83)",
        "borderRadius": 18,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "margin": 7,
        "width": "90%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="rgb(158, 158, 158)"
      rejectResponderTermination={true}
      style={
        Object {
          "color": "white",
          "fontSize": 17,
          "margin": 7,
          "width": "60%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(193, 73, 83)",
          "borderRadius": 18,
          "height": undefined,
          "margin": 10,
          "opacity": 1,
          "padding": 10,
          "width": "25%",
        }
      }
      testID="button"
    >
      <Text
        style={
          Object {
            "color": "rgb(255, 255, 255)",
            "fontSize": 15,
          }
        }
      >
        Add
         
      </Text>
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "rgb(193, 73, 83)",
        "fontSize": 12,
        "height": 15,
        "margin": 0,
        "paddingLeft": 3,
        "width": "75%",
      }
    }
  >
     
  </Text>
</View>
`;
